src/mongo.js

```diff
@@ -0,0 +1,75 @@
+function extend(Mongo: Object) {
+  if (!Mongo) {
+    throw new Error('Mongo can\'t should be truthy!');
+  }
+
+  if (!Mongo.Collection || typeof Mongo.Collection !== 'function') {
+    throw new Error('Mongo.Collection is not a prototypable constructor!');
+  }
+
+  Object.assign(Mongo.Collection.prototype, {
+    updateAsync(selector, modifier, options) {
+      return new Promise((resolve, reject) => {
+        this.update(selector, modifier, options, (error, numUpdated) => {
+          if (error) {
+            reject(error);
+          } else {
+            resolve(numUpdated);
+          }
+        });
+      });
+    },
+
+    updateManyAsync(selector, modifier, options) {
+      return this.updateAsync(selector, modifier, Object.assign({}, options, { multi: true }));
+    },
+
+    insertAsync(document) {
+      return new Promise((resolve, reject) => {
+        this.insert(document, (error, id) => {
+          if (error) {
+            reject(error);
+          } else {
+            resolve(id);
+          }
+        });
+      });
+    },
+
+    upsertAsync(query, modifier) {
+      return new Promise((resolve, reject) => {
+        this.upsert(query, modifier, (error, result) => {
+          if (error) {
+            reject(error);
+          } else {
+            resolve(result);
+          }
+        });
+      });
+    },
+
+    removeAsync(selector) {
+      return new Promise((resolve, reject) => {
+        this.remove(selector, (error, numRemoved) => {
+          if (error) {
+            reject(error);
+          } else {
+            resolve(numRemoved);
+          }
+        });
+      });
+    },
+
+    _ensureNoIndex(selector) {
```

Comment:
If we're exporting this as part of a package intended for reuse, we should remove the underscore prefix

Reply:
Should we, though? The underscore prefix is for consistency with the peer `_dropIndex` method.

Reply:
Yeah, I understand the reasoning, and that's why I was okay with that name before when this code just lived inside of our own monolithic codebase.

But using this naming here introduces a different inconsistency that I think is more important to resolve -- underscore prefixes mean "don't use this externally," so it doesn't make sense for us to export a method like that in an NPM package. I value not violating that rule over keeping the method names symmetric.